File: models/Report.js
const mongoose = require('mongoose');

const reportSchema = new mongoose.Schema({
  subject: {
    type: String,
    required: true,
    enum: ['water', 'electric', 'sanitation', 'roads', 'animals', 'others']
  },
  subjectLabel: {
    type: String,
    required: true
  },
  title: {
    type: String,
    required: true
  },
  details: {
    type: String,
    required: true
  },
  location: {
    type: String,
    required: true
  },
  fileUrl: {
    type: String
  },
  status: {
    type: String,
    enum: ['Pending', 'In Progress', 'Resolved'],
    default: 'Pending'
  },
  reply: {
    type: String,
    default: ''
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Report', reportSchema);

File: src/main/java/com/weconnect/model/Report.java
package com.weconnect.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@Document(collection = "reports")
public class Report {
    @Id
    private String id;
    private String subject;
    private String subjectLabel;
    private String title;
    private String details;
    private String location;
    private String fileUrl;
    private String status = "Pending";
    private String reply = "";
    private String userId;
    private Date createdAt = new Date();
    private Date updatedAt = new Date();
}
